apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/transaction/DexOfferTransaction.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright © 2018-2019 Apollo Foundation
+ */
+package com.apollocurrency.aplwallet.apl.exchange.transaction;
+
+import com.apollocurrency.aplwallet.apl.core.account.Account;
+import com.apollocurrency.aplwallet.apl.core.account.LedgerEvent;
+import com.apollocurrency.aplwallet.apl.core.app.Transaction;
+import com.apollocurrency.aplwallet.apl.core.transaction.TransactionType;
+import com.apollocurrency.aplwallet.apl.core.transaction.messages.AbstractAttachment;
+import com.apollocurrency.aplwallet.apl.core.transaction.messages.DexOfferAttachment;
+import com.apollocurrency.aplwallet.apl.exchange.model.DexOffer;
+import com.apollocurrency.aplwallet.apl.exchange.service.DexService;
+import com.apollocurrency.aplwallet.apl.util.AplException;
+import org.json.simple.JSONObject;
+
+import javax.enterprise.inject.spi.CDI;
+import javax.inject.Singleton;
+import java.nio.ByteBuffer;
+
+@Singleton
+public class DexOfferTransaction extends DEX {
+
+    private DexService dexService = CDI.current().select(DexService.class).get();
+
+    @Override
+    public byte getSubtype() {
+        return TransactionType.SUBTYPE_DEX_OFFER;
+    }
+
+    @Override
+    public LedgerEvent getLedgerEvent() {
+        return LedgerEvent.TRANSACTION_FEE;
+    }
+
+    @Override
+    public AbstractAttachment parseAttachment(ByteBuffer buffer) throws AplException.NotValidException {
+        return new DexOfferAttachment(buffer);
+    }
+
+    @Override
+    public AbstractAttachment parseAttachment(JSONObject attachmentData) throws AplException.NotValidException {
+        return new DexOfferAttachment(attachmentData);
+    }
+
+    @Override
+    public void validateAttachment(Transaction transaction) throws AplException.ValidationException {
+        DexOfferAttachment attachment = (DexOfferAttachment) transaction.getAttachment();
+
+        if (attachment.getOfferCurrency() == attachment.getPairCurrency()) {
+            throw new AplException.NotCurrentlyValidException("Invalid Currency codes: " + attachment.getOfferCurrency() + " / " + attachment.getPairCurrency());
+        }
+        //TODO add validation cases.
```

Comment:
Validation should be added ASAP, because some early transactions can become invalid therefore forks can occurred